plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/IngestActionFilter.java

```diff
@@ -61,7 +63,8 @@ public void apply(String action, ActionRequest request, ActionListener listener,
             processIndexRequest(action, listener, chain, (IndexRequest) request, pipelineId);
         } else if (request instanceof BulkRequest) {
             BulkRequest bulkRequest = (BulkRequest) request;
-            processBulkIndexRequest(action, listener, chain, bulkRequest, pipelineId, bulkRequest.requests().iterator());
+            BulkRequestSource state = new BulkRequestSource(bulkRequest, listener);
```

Comment:
can we cast the listener to ActionListener<BulkResponse> and suppress the unchecked warning for that specific line?